[WIP] Remove links to talk pages from old schedule views

rahmonov opened the pull request:
Please, let me know if I am on the right path.

P.S. I couldn't run the tests locally. There is this error coming up: `You have offline compression enabled but key "1354bed19bb78fa1bb9f122d6ee27708" is missing from offline manifest. You may need to run "python manage.py compress".` Running "python manage.py compress" did not help. Any ideas how I can fix it?

<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

This PR closes/references issue #281 . 

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
1. Release multiple schedules. Go to an old version. None of the talks should have a link.
2. Delete some submission. It should be displayed as "[deleted]"

## Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/9026972/33185397-25d0bcd0-d0a4-11e7-9849-d94fe371742b.png)

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- It is ok to not check all boxes! We just want to know if we need to do any work after merging the PR. -->

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My change is listed in the CHANGELOG.rst if appropriate.

rixx commented:
What command did you run when you encountered the issue, exactly? And have you installed all the development dependencies?

rixx commented on src/pretalx/agenda/templates/agenda/schedule.html:
```diff
@@ -72,9 +74,15 @@ <h3 id="{{ day.start.date|date:"c" }}">
-                                            <span class="talk-speakers">({{ talk.submission.display_speaker_names }})</span><br>
+
+                                        {% if talk.submission.is_deleted %}
+                                          <span class="talk-title">[deleted]</span>
```
This would work better as a translatable string.

rixx commented on src/pretalx/agenda/views/schedule.py:
```diff
@@ -62,11 +62,11 @@ def get_context_data(self, *args, **kwargs):
                     'name': room.name,
                     'talks': [talk for talk in talks
-                              if talk.start.astimezone(tz).date() == current_date.date() and talk.room_id == room.pk],
+                              if talk.start and talk.start.astimezone(tz).date() == current_date.date() and talk.room_id == room.pk],
```
Good catch!

rahmonov commented:
@rixx please take a look. I removed deleted submissions from rss feed. Added a couple of tests. Made "[deleted]" a translatable string. 

For some reason, when I tried to rebase, git created the same commit once again. That's why there are 4 commits instead of 3. I surely need to have a read on this topic.

P.S. I encounter that error when I just do "pytest" and some tests fail by saying "You have offline compression enabled but key "1354bed19bb78fa1bb9f122d6ee27708" is missing from offline manifest. You may need to run "python manage.py compress". I will try to fix it and let you know if I do.

rixx commented:
Thank you very much! :rocket:

rahmonov commented:
🚀🚀🚀
